src/ui/components/mixins/notebook-snapshot.js

```diff
@@ -0,0 +1,130 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+function generateTaskForm(url) {
+    var taskForm = {
+        name: "Create a Notebook Entry",
+        hint: "Please select a Notebook",
+        sections: [{
+            rows: [
+                {
+                    name: 'Entry',
+                    key: 'entry',
+                    control: 'textarea',
+                    required: false,
+                    cssClass: "l-textarea-sm"
+                },
+                {
+                    name: 'Snap Preview',
+                    key:"snapPreview",
+                    control: "snap-view",
+                    cssClass: "l-textarea-sm",
+                    src: url
+                },
+                {
+                    name: 'Save in Notebook',
+                    key: 'notebook',
+                    control: 'locator',
+                    validate: validateLocation
+                }
+            ]
+        }]
+    };
+
+    var overlayModel = {
+        title: taskForm.name,
+        message: 'Notebook Snapshot',
+        structure: taskForm,
+        value: {'entry': ""}
+    };
+
+    function validateLocation(newParentObj) {
+        return newParentObj.model.type === 'notebook';
+    }
+
+    return overlayModel;
+}
+
+function saveSnapShot(dialogService, imageUrl, imageType, imageSize, embedObject) {
+    let taskForm = generateTaskForm(imageUrl);
+
+    dialogService.getDialogResponse(
+        'overlay-dialog',
+        taskForm,
+        () => taskForm.value
+    ).then(options => {
+        let snapshotObject = {
+            src: options.snapPreview || imageUrl,
+            type: imageType,
+            size: imageSize
+        };
+
+        options.notebook.useCapability('mutation', function (model) {
+            var date = Date.now();
+
+            model.entries.push({
+                id: 'entry-' + date,
+                createdOn: date,
+                text: options.entry,
+                embeds: [{
+                    name: embedObject.name,
+                    cssClass: embedObject.cssClass,
+                    type: embedObject.id,
+                    id: 'embed-' + date,
+                    createdOn: date,
+                    snapshot: snapshotObject
+                }]
+            });
+        });
+    });
+}
+
+export default {
+    inject: ['openmct', 'domainObject'],
+    methods: {
+        takeSnapshot(DOMElement) {
```

Comment:
Camel case applies even when the first word is an acronym. Helps to distinguish instance objects from classes etc.

Reply:
Sorry, I overlooked this. Fixed in latest commit.